File: programmers_problmes/python3/네트워크_lv3.py
def solution(n, computers):
    answer = 0
    visit = []
    for current_node in range(n):
        if current_node in visit:
            continue
        
        answer += 1
        queue = [current_node]
        visit += [current_node]
        
        while(queue):
            current_node = queue.pop(0)
            for search_node in range(n):
                if search_node not in visit and computers[current_node][search_node] == 1:
                    queue.append(search_node)
                    visit += [search_node]

    
    
    return answer

File: programmers_problmes/kotlin/네트워크_lv3.kt
var networkCount = 0
var computerNetwork = arrayListOf<IntArray>()

class Solution {
    fun solution(n: Int, computers: Array<IntArray>): Int {

        var sets = mutableSetOf<List<Int>>()
        computerNetwork = computers.toCollection(ArrayList())


        for(i in 0 until n) {
            val route = dfs(i, n, arrayListOf<Int>(), MutableList(n){false})
            sets.add(route.sorted())
        }

        println(sets)
        return sets.size
    }

    fun dfs(startNode: Int, n: Int, comb: ArrayList<Int>, visit: MutableList<Boolean>): ArrayList<Int> {

        visit[startNode] = true
        comb.add(startNode)

        for(i in 0 until n) {
            if(visit[i] == true || computerNetwork[startNode][i] == 0) continue
            dfs(i , n, comb, visit)
        }

        return comb

    }
}